### ISSUE TITLE
Need test case for MIAF 7.4.2

### ISSUE THREAD

**Issue by aklemets** [gpac/ComplianceWarden#4]
I could not find a test case for MIAF paragraph 7.4.2:

> Thumbnail or auxiliary sequences shall conform either to the requirements of the same MIAF codec profile as the MIAF image sequence or video to which it is linked, or to the requirements of a format defined in clause 9

**Comment by rbouqueau** [comment]
You are right. It was internally marked as done but the test is missing. I'll add it when doing the next batch of modifications.

**Comment by rbouqueau** [comment]
I still miss clarifications about what a "MIAF codec profile" is. If I believe the next TuC (w18851 section 1.1.1) then it is equivalent to a MIAF Profile. MIAF profiles are indicated in the 'ftyp' box i.e. once per file.

For a MIAF content with several tracks (each of them having a thumbnail or auxiliary sequence), how can we know which track refers to which MIAF Profile?

For sure the compliance checker can look for brands, and try to see if the supported brands of a track also applies to its thmb/auxl. But it means we could report false negative.

Opinion on this?

CC @cconcolato

**Comment by cconcolato** [comment]
Interesting question. 

If there are multiple MIAF tracks, with different MIAF profiles, I would expect the ftyp box to list all the profiles, but it might be tricky to associate a brand with a track. An alternative would be to use the TrackTypeBox, but I'm not sure what its status is in MIAF. I don't think you can rely on it. 

In summary, I don't think you should use brands to determine the MIAF codec profile. Rather you should check that the codec, profile and level are the same, but that might give false negatives.

**Comment by rbouqueau** [comment]
I will implement it with one track. Let's try to clarify the multiple track case, at the moment it seems non enforcable.

**Comment by rbouqueau** [comment]
@cconcolato Do we have samples for that or not yet?

**Comment by cconcolato** [comment]
No, but that shouldn't be difficult to create with MP4Box, no?

**Comment by rbouqueau** [comment]
I don't see exactly how to do it with the current GPAC code. We could add a 'ttyp' box on any IFF file GPAC creates with a track (I see that we don't add (comply with ?) the 'msf1' brand. Did you have another idea in mind?

**Comment by cconcolato** [comment]
What I meant is that:
* we can create or find valid files:
  * with master and thumbnail (or auxiliary) images with the same profile
  * with a master image in a format and a thumbnail image (or auxiliary image) in JPEG 
* we can create or find invalid files: e.g. a thumbnail that requires a profile exceeding the one indicated in the master image.
